actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -740,8 +729,8 @@ def apply_form_for_options!(record, object, options) # :nodoc:
       #     form_with(**options.merge(builder: LabellingFormBuilder), &block)
       #   end
       def form_with(model: nil, scope: nil, url: nil, format: nil, **options, &block)
-        options[:allow_method_names_outside_object] = true
-        options[:skip_default_ids] = !form_with_generates_ids
+        options.with_defaults!(allow_method_names_outside_object: true)
```

Comment:
Let's avoid this `with_defaults!` this method is too confusing.

```suggestion
        options = { allow_method_names_outside_object: true, skip_default_ids: !form_with_generates_ids }.merge!(options)
```

is easier to understand.